### ISSUE TITLE
unresolved-symbol for all-lowercase class name

### ISSUE THREAD

**Issue by vemv** [clj-kondo/clj-kondo#1362]
The following code:

```clj
(def ^name.fraser.neil.plaintext.diff_match_patch dmp (diff_match_patch.))
```

will result in `error: Unresolved symbol: name.fraser.neil.plaintext.diff_match_patch`

(I place the fully-qualified type hint for 1.7 compat)

`name.fraser.neil.plaintext.diff_match_patch` is a valid class name as found in `[org.clojars.brenton/google-diff-match-patch "0.1"]`

**Environment**

`[clj-kondo "2021.08.06"]`, JVM at runtime

**Comment by borkdude** [comment]
Thanks!

Can you explain what you meant with 1.7 compat?

Just a note to self, the type hint in that position on a var can take care of reflection warnings, e.g.:

```
user=> (set! *warn-on-reflection* true)
true
user=> (def x nil)
#'user/x
user=> (defn foo [] (.length x))
Reflection warning, NO_SOURCE_PATH:1:14 - reference to field length can't be resolved.
#'user/foo
user=> (def ^String x nil)
#'user/x
user=> (defn foo [] (.length x))
#'user/foo
```

**Comment by vemv** [comment]
> Can you explain what you meant with 1.7 compat?

Clojure <= 1.7 generally needs type hints to be fully qualified, else they often will be wrong when consumed from another ns

**Comment by borkdude** [comment]
You mean fully qualified vs the imported classname right?

**Comment by borkdude** [comment]
The reason this currently is a false positive in clj-kondo has to do with the heuristic of how classes are detected. E.g. this one works:

```
(def ^name.fraser.neil.plaintext.Diff_match_atch dmp (diff_match_patch.))
```

because of the capital letter.

**Comment by vemv** [comment]
> You mean fully qualified vs the imported classname right?

Here's a description of the behavior pre clojure 1.8 https://github.com/metrics-clojure/metrics-clojure/issues/66

i.e. it didn't matter if one `:import`ed a class

**Comment by vemv** [comment]
🍻
